igvm/hypervisor.py

```diff
@@ -555,15 +556,77 @@ def vm_block_device_name(self):
         the guest OS"""
         return 'vda1'
 
-    def migrate_vm(
-        self, vm, target_hypervisor, offline, offline_transport, transaction,
-        no_shutdown,
+    def check_migrate_vm(
+        self, vm: VM, offline: bool, offline_transport: str,
+        disk_size: int = 0,
     ):
-        if offline_transport not in ['netcat', 'drbd']:
+        if offline_transport not in ['netcat', 'drbd', 'xfs']:
             raise StorageError(
                 'Unknown offline transport method {}!'
                 .format(offline_transport)
             )
+        disk_size = disk_size if 0 < disk_size else 0
+        if not (offline and offline_transport == 'xfs') and 0 < disk_size:
+            raise StorageError(
+                'disk_size can be applied only with offline transport xfs!'
+            )
+        allocated_space = vm.dataset_obj['disk_size_gib'] - vm.disk_free()
+        if 0 < disk_size < allocated_space:
+            raise StorageError(
+                'disk_size is lower than allocated space: {} < {}!'
+                .format(disk_size, allocated_space)
+            )
+
+    def migrate_disk_size(
+        self, vm: VM, offline: bool, offline_transport: str,
+        disk_size: int = 0,
+    ) -> int:
+        self.check_migrate_vm(vm, offline, offline_transport, disk_size)
+        disk_size = disk_size if 0 < disk_size else 0
+        return disk_size or vm.dataset_obj['disk_size_gib']
+
+    def migrate_change_disk_size(
+        self, vm: VM, offline: bool, offline_transport: str,
+        transaction, disk_size: int = 0,
+    ):
```

Comment:
Same here. The name does not really say what the method is doing IMHO. Maybe we use something like `apply_new_disk_size` or `save_new_disk_size` and a doc string.

Reply:
Changed. Looks better?

Reply:
Why not `apply_new_vm_disk_size` - this is more close to our every day English than `apply_vm_new_disk_size` ?

Reply:
I wanted to make it consistent with `vm_new_disk_size` -> `apply_vm_new_disk_size`. First vm indicates the scope.

Maybe, `_vm_apply_new_disk_size` would sound better. The method looks private to me.

Reply:
Yes I think that is much better.